source/daplink/interface/target_flash.c

```diff
@@ -172,11 +172,20 @@ static error_t target_flash_erase_chip(void)
 static uint32_t target_flash_program_page_min_size(uint32_t addr)
 {
     uint32_t size = 256;
-    util_assert(target_device.sector_size >= size);
+    util_assert(target_flash_erase_sector_size(addr) >= size);
     return size;
 }
 
 static uint32_t target_flash_erase_sector_size(uint32_t addr)
 {
+#ifdef SECTOR_INFO_AVAILABLE
+    int num_sectors = sizeof(sectors_info)/sizeof(sectors_info[0]);
+    for (int sector_index = 0; sector_index < (num_sectors-1); sector_index+=2) {
```

Comment:
This information should be part of target_device so all device related information is encapsulated in this object. Potentially, target_device.sector_size should be replaced with the sector info.